packages/react/src/components/Menu/MenuItem.tsx

```diff
@@ -342,31 +347,40 @@ class MenuItem extends AutoControlledComponent<ReactProps<MenuItemProps>, MenuIt
     _.invoke(this.props, 'onFocus', e, this.props)
   }
 
-  private closeAllMenus = (e, focusNextParent: boolean) => {
-    const { menu, inSubmenu } = this.props
+  private isSubmenuOpen = (): boolean => {
+    const { menu } = this.props
     const { menuOpen } = this.state
-    if (menu && menuOpen) {
-      this.trySetMenuOpen(false, e, () => {
-        if (!inSubmenu && (!focusNextParent || this.props.vertical)) {
-          focusAsync(this.itemRef.current)
-        }
-      })
+
+    return menu && menuOpen
+  }
+
+  private closeAllMenus = (e, focusNextParent: boolean) => {
```

Comment:
nit: maybe give types `React.SyntheticEvent` to the `e` here and anywhere else since you in the area.

Reply:
Unfortunately, `event` object is passed through accessibility action handlers with a type of `Event` because behaviors are targeted not only React framework

Reply:
then make it `Event` :D